lib/iris/tests/__init__.py

```diff
@@ -733,13 +749,27 @@ def check_graphic(self, tol=None):
                     # There is no difference between the actual and expected
                     # result, so remove the actual result file.
                     os.remove(result_fname)
-                
+
             if _DISPLAY_FIGURES:
                 plt.show()
 
         finally:
             plt.close()
 
+    def check_graphic(self, tol=None):
+        """
+        Checks that the image hash for the current matplotlib figure matches
+        the expected image hash for the current test.
+
+        """
+        fname = os.path.join(os.path.dirname(__file__),
+                             'results', 'imagerepo.lock')
+        lock = filelock.FileLock(fname)
+        # The imagerepo.json file is a critical resource, so ensure thread
+        # safe read/write behaviour via platform independent file locking.
+        with lock.acquire(timeout=600):
```

Comment:
@marqh A suitably high timeout ...

Having the graphics tests fighting over the `imagerepo.lock` doesn't seem to impact the overall running time of the tests in travis ... but it's hard to tell due to the variable loads on travis.

I don't think this is a problem, but we could choose to only acquire a lock if the `IRIS_TEST_CREATE_MISSING` environment variable is set - as this means there will be potential writing to the `imagerepo.json` and this shouldn't be applicable to test runs within travis but only on the desktop of the developer.